gp-includes/things/project.php

```diff
@@ -342,5 +342,22 @@ public function duplicate_project_contents_from( $source_project ){
 		}
 	}
 
+	/**
+	 * Deletes a project and all of sub projects, translations, translation sets, originals and glossaries.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return bool
+	 */
+	public function delete() {
+		GP::$project->delete_many( array( 'parent_project_id' => $this->id ) );
+		
+		GP::$translation_set->delete_many( array( 'project_id' => $this->id ) );
+
+		GP::$original->delete_many( array( 'project_id' => $this->id ) );
```

Comment:
I'd say no, simply because we're not going to undo the previous steps if something fails.

There could also be valid instances of a step "failing" (like maybe the project doesn't have any glossaries or translation sets) but we would still want to continue on.